### PR TITLE
Add 3D spotlight and spotpoint gallery examples

### PR DESCRIPTION
**Pull request by rocky**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /node_modules
-package-lock.json
+/tmp
```
Comment (TiagoCavalcanteTrindade) [inline .gitignore:2]: Is it something like .DS_Store?

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /node_modules
-package-lock.json
```
Comment (TiagoCavalcanteTrindade) [inline .gitignore:2]: I don't think this file is needed.

File: .gitignore (hunk 1 above)
Comment (rocky) [reply .gitignore:2]: It is my own convention for a place to put temporary files specific to this project. It is like a local `/tmp`

File: .gitignore (hunk 2 above)
Comment (rocky) [reply .gitignore:2]: That line is just moved around to alphabetic order. It has been added at the end of the file. My editor seems to like to order lines in `.gitignore`

Comment (TiagoCavalcanteTrindade) [reply .gitignore:2]: Oh, ok.

File: .gitignore (hunk 1 above)
Comment (TiagoCavalcanteTrindade) [reply .gitignore:2]: Ok.

File: examples/gallery/objects-in-splotlight.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+					 opacity: 1.0,
+					 coords: [[[1.0, 2.0, -1.0]],[[2.0, 3.0, 1.0]]]
+					},
+					{type: 'uniformPolyhedron',
+					 subType: "dodecahedron",
+					 color: [1.0, 1.0, 1.0],
+					 opacity: 1.0,
+					 coords: [[[3.0,-1.0, 0.0]]]}],
+				lighting: [
+					{type: 'directional',
```
Comment (TiagoCavalcanteTrindade) [inline examples/gallery/objects-in-splotlight.html:38]: Maybe change the file name?

Comment (rocky) [reply examples/gallery/objects-in-splotlight.html]: By all means change the name to whatever makes sense. Thanks

And the reformatting is fine too. My Emacs isn't reformatting this HTML automatically. If tweaking the EditorConfig file will help that would be great.

Lastly, some of these mirror (or are better examples) of what is in the wiki. So the wiki examples and PNG's should be updated and links to the on-line versions I think would help.

**Pull request merged**